common/scala/src/main/scala/whisk/http/BasicHttpService.scala

```diff
@@ -108,7 +108,7 @@ trait BasicHttpService extends Directives {
     extract { req =>
       val tid =
         req.request.headers
-          .find(_.name == TransactionId.generatorConfig.header)
+          .find(_.name.toLowerCase == TransactionId.generatorConfig.header.toLowerCase)
```

Comment:
Great catch! akka-http normalizes everything to lowercase, so you can use:

```scala
.find(_.is(TransactionId.generatorConfig.header.toLowerCase))
```

We should also cache the lowercased value of the configured header somewhere, needs to be computed only once.